src/stan/model/finite_diff_grad.hpp

```diff
@@ -35,6 +38,7 @@ namespace stan {
       std::vector<double> perturbed(params_r);
       grad.resize(params_r.size());
       for (size_t k = 0; k < params_r.size(); k++) {
+        interrupt();
```

Comment:
How often is the interrupt supposed to be called? This is once per log density evaluation but I don't think that's how ti works with HMC (?)